pageserver/src/tenant/timeline/offload.rs

```diff
@@ -33,6 +34,21 @@ pub(crate) async fn offload_timeline(
     debug_assert_current_span_has_tenant_and_timeline_id();
     tracing::info!("offloading archived timeline");
 
+    if !timeline.remote_client.no_pending_work() {
+        return Err(OffloadError::RemoteStorage(anyhow!(
+            "cannot offload timeline with pending uploads"
+        )));
+    }
+
+    // Drain again in case of race condition (i.e., some background tasks add new uploads right after we checked)
```

Comment:
this doesn't prevent race conditions, we could have a background upload right after this.

Reply:
Yes, so I also set the upload mode to be `FreezeAndFlush` to ensure everything gets eventually uploaded. Doing a wait-for-completion early can reduce the shutdown time.

Reply:
In that case the comment is wrong, but I also think it is not really needed: shutdown shouldn't take that long in the first place.